src/inmanta/module.py

```diff
@@ -276,14 +320,155 @@ def get_file_for_version(self, repo: str, tag: str, file: str) -> str:
+            namespace_package: str = self.get_namespace_package_name(module_name)
+            raise InvalidModuleException(f"{python_package} does not contain a {namespace_package} module.")
+        return self.from_path(project, path)
+
+    def path_for(self, name: str) -> Optional[str]:
+        """
+        Returns the path to the module root directory. Should be called prior to configuring the module finder for v1 modules.
+        """
+        if name.startswith(ModuleV2.PKG_NAME_PREFIX):
+            raise Exception("PythonRepo instances work with inmanta module names, not Python package names.")
+        package: str = self.get_namespace_package_name(name)
+        mod_spec: Optional[Tuple[str, Loader]] = env.ProcessEnv.get_module_file(package)
+        if mod_spec is None:
+            return None
+        init, mod_loader = mod_spec
+        if isinstance(mod_loader, loader.PluginModuleLoader):
+            # TODO: test this behavior
+            # module was found in the environment but it is associated with the v1 module loader
+            return None
+        pkg_installation_dir = os.path.abspath(os.path.dirname(init))
+        if os.path.exists(os.path.join(pkg_installation_dir, ModuleV2.MODULE_FILE)):
+            # normal install: __init__.py is in module root
+            return pkg_installation_dir
+        else:
+            # editable install: __init__.py is in `inmanta_plugins/<mod_name>`
+            module_root_dir = os.path.normpath(os.path.join(pkg_installation_dir, os.pardir, os.pardir))
+            if os.path.exists(os.path.join(module_root_dir, ModuleV2.MODULE_FILE)):
+                return module_root_dir
+        raise Exception(f"Invalid module: found plugins package but the module has no {ModuleV2.MODULE_FILE}.")
+
+    @classmethod
+    def from_path(cls, project: Optional["Project"], path: str) -> "ModuleV2":
+        return ModuleV2(project, path, is_editable_install=os.path.exists(os.path.join(path, const.PLUGINS_PACKAGE)))
+
+    def _get_module_name(self, module_spec: List[InmantaModuleRequirement]) -> str:
+        module_name: str = super()._get_module_name(module_spec)
+        if module_name.startswith(ModuleV2.PKG_NAME_PREFIX.replace("-", "_")):
+            raise Exception("PythonRepo instances work with inmanta module names, not Python package names.")
+        return module_name
+
+
+class ModuleRepo(ModuleSource["ModuleV1"]):
+    def clone(self, name: str, dest: str) -> bool:
+        raise NotImplementedError("Abstract method")
+
+    def install(self, project: "Project", module_spec: List[InmantaModuleRequirement]) -> Optional["ModuleV1"]:
+        module_name: str = self._get_module_name(module_spec)
+        path: Optional[str] = self.path_for(module_name)
+        if path is not None:
+            return self.from_path(project, path)
+        else:
+            return ModuleV1.install(project, module_name, module_spec, install_mode=project.install_mode)
```

Comment:
Why not move this method from ModuleV1 to here?

Reply:
Done. This required a bit of a refactor of the v1 module source: 0b70c9e.